spec/helpers/glossary_helper_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+RSpec.describe GlossaryHelper do
+  def sanitize(doc)
+    doc.tap{ doc.css('img').each(&:remove) }
+  end
+
+  describe '#link_glossary_terms' do
+    let(:html){ '<p>glossary helper spec</p>' }
+    let(:html_with_repeated_term){ '<p>glossary helper spec with glossary term</p>' }
+    it 'should do nothing when the document contains no terms' do
+      doc = Nokogiri::HTML.fragment(html)
+
+      helper.link_glossary_terms(doc)
+      expect(sanitize(doc).to_html).to eq(html)
```

Comment:
`helper.link_glossary_terms(doc)` and `sanitize(doc).to_html` could also be extracted into `let`s for DRYness and legibility
```
let(:doc) { Nokogiri::HTML.fragment(html) }
let(:link_terms) { helper.link_glossary_terms(doc) }
subject(:sanitized_doc) { sanitize(doc).to_html }

it 'should do nothing when the document contains no terms' do
  link_terms
  expect(sanitized_doc).to eq(html)
end

it 'should replace glossary terms in the document with links' do
  GlossaryTerm.create!(name: 'glossary')
  GlossaryTerm.create!(name: 'Spec')
  link_terms

  expect(sanitized_doc).to eq(
    '<p><a href="/glossary/glossary" class="glossary-term">glossary</a> helper <a href="/glossary/spec" class="glossary-term">spec</a></p>'
  )
end
```